lib/file.js

```diff
@@ -306,27 +347,52 @@ class File extends EventEmitter {
     }
 
     // Supported formats:
+    // Old format:
     // https://mega.nz/#!file_handker
     // https://mega.nz/#!file_handker!file_key
     // https://mega.nz/#F!folder_handker
     // https://mega.nz/#F!folder_handker!folder_key
     // https://mega.nz/#F!folder_handker!folder_key!file_handle
+    // New format (2020):
+    // https://mega.nz/file/file_handker
+    // https://mega.nz/file/file_handker#file_key
+    // https://mega.nz/folder/folder_handker
+    // https://mega.nz/folder/folder_handker#folder_key
 
     const url = parse(opt)
-    if (url.hostname !== 'mega.nz' && url.hostname !== 'mega.co.nz') throw Error('Invalid URL: wrong hostname')
+    if (url.hostname !== 'mega.nz' && url.hostname !== 'mega.co.nz') { throw Error('Invalid URL: wrong hostname') }
     if (!url.hash) throw Error('Invalid URL: no hash')
 
-    const split = url.hash.split('!')
-    if (split[0] !== '#' && split[0] !== '#F') throw Error('Invalid URL: format not recognized')
-    if (split.length <= 1) throw Error('Invalid URL: too few arguments')
-    if (split.length >= (split[0] === '#' ? 4 : 5)) throw Error('Invalid URL: too many arguments')
-
-    return new File({
-      downloadId: split[1],
-      key: split[2],
-      directory: split[0] === '#F',
-      loadedFile: split[3]
-    })
+    if (url.path.match(/\/(file|folder)\//) !== null) {
+      // new format
+      const fileHandker = url.path.substring(
+        url.path.lastIndexOf('/') + 1,
+        url.path.length + 1
+      )
+      const fileKey = url.hash.substr(1)
+
+      if ((fileHandker && !fileKey) || (!fileHandker && fileKey)) throw Error('Invalid URL: too few arguments')
+
+      return new File({
+        downloadId: fileHandker,
+        key: fileKey,
+        directory: url.path.indexOf('/folder/') >= 0,
+        loadedFile: null
```

Comment:
@qgustavor is this correct? Wasn't sure for what this is used.

Reply:
`loadedFile` is for selecting nodes in folders: when you open a shared folder then click on a folder the handler of it will be stored in the URL.

Reply:
Oh okay, I've checked the URL for shared files within folders and it was one of the new /file/ URL.

Reply:
I just tested it, seems the new syntax is `https://mega.nz/folder/FOLDERHANDLER#KEY/file/FILEHANDLER` and `https://mega.nz/folder/FOLDERHANDLER#KEY/folder/FOLDERHANDLER`.

Reply:
Ohh okay, thought it meant a shared file nested inside a folder, so I guess the code needs some rework for this case.